data/npc/lib/npc.lua

```diff
@@ -56,68 +56,44 @@ function doNpcSellItem(cid, itemid, amount, subType, ignoreCap, inBackpacks, bac
 end
 
 local func = function(cid, text, type, e, pcid)
-	if isPlayer(pcid) then
-		doCreatureSay(cid, text, type, false, pcid, getCreaturePosition(cid))
-		e.done = TRUE
+	if Player(pcid):isPlayer() then
+		local creature = Creature(cid)
+		creature:say(text, type, false, pcid, creature:getPosition())
+		e.done = true
 	end
 end
 
 function doCreatureSayWithDelay(cid, text, type, delay, e, pcid)
-	if isPlayer(pcid) then
-		e.done = FALSE
+	if Player(pcid):isPlayer() then
+		e.done = false
 		e.event = addEvent(func, delay < 1 and 1000 or delay, cid, text, type, e, pcid)
 	end
 end
 
-function doPlayerTakeItem(cid, itemid, count)
```

Comment:
Add this to compat, doPlayerTakeItem is commonly used in NPC scripts across datapacks.